### ISSUE TITLE
[Deprecation] Explore deprecating the staking `chill` extrinsic

### ISSUE THREAD

**Issue by emostov** [paritytech/polkadot-sdk#314]
paritytech/substrate#8920 Introduces the `chill_other` extrinsic which allows anyone to chill a staker below `Min{Nominator, Validator}Bond`. The extrinsic also allows a controller to chill itself regardless of bonded funds, making the `chill` extrinsic redundant; thus we should consider deprecating the `chill` extrinsic.

```[tasklist]
### Deprecation steps
- [ ] Soft deprecation and removal from the codebase
- [ ] Update examples and tutorials
- [ ] Hard deprecation by adding a deprecate compiler warning
- [ ] Deprecation is noted in the release log
- [ ] Removal version is planned
- [ ] Deprecated feature is removed from the code base
- [ ] Removal of the feature is noted in the release log
```

**Comment by Captain1653** [comment]
I looked into this issue regarding the updates for `pallet-staking`, but since this pallet is already marked as deprecated (as seen in #11940 ), there is no longer any point in making changes here, and this issue can be closed.

However, I noticed that `pallet-async-staking` has succeeded the old staking pallet, and it still contains both the `chill` and `chill_other` methods. 

If it's necessary  to properly deprecate this in `async-staking`, then there is  the formal process outlined in the [deprecation checklist](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/DEPRECATION_CHECKLIST.md). Perhaps a separate issue should be created for this task.

If it's still relevant **(though I'm not entirely sure it is, given the stability of the current API)**, I can take it on and handle the initial steps (marking the method as deprecated and updating internal usages/tests/examples/docs). I understand that coordinating with frontend teams and managing downstream integration impacts would be handled administratively by the maintainers.

**Comment by sigurpol** [comment]
Agreed on the first half: pallet-staking is deprecated so there is nothing to do there.

On carrying the idea over to pallet-staking-async, I'd argue against it, and I think this issue should be closed as won't-do rather than reopened against the new pallet.
It is true that a controller can chill itself through `chill_other` and so we could deprecate `chill` on paper,  but in practice (see e.g.  https://assethub-polkadot.subscan.io/extrinsic?page=1&time_dimension=date&module_call=%5B%7B%22module%22%3A%22staking%22%2C%22call%22%3A%22chill%22%7D%5D)
`chill` is the call people actually use it commonly, so deprecating while wildly used  with real low benefit is not worth it imho.

Also I hink the `chill` family has 3 calls that expressed different intents:
- chill : I am stopping
- chill_other: this stake no longer qualifies
- chill_inactive: chill an inactive validator

Pragmatically I would just keep things as they are now.

Thanks for digging into it though, much appreciated 🙇 

Closing -  @kianenigma , @Ank4n  you are more than welcome to re-open it if you think it should be tackled instead
